src/tools/debugger/stepping/Stepping.java

```diff
@@ -0,0 +1,42 @@
+package tools.debugger.stepping;
+
+import tools.SourceCoordinate.FullSourceCoordinate;
+/**
+ *
+ * Manage the actor message stepping.
+ *
+ */
+public class Stepping {
+
+  public enum SteppingType{
+    STEP_INTO,
+    STEP_OVER,
+    STEP_RETURN
+  }
+
+  private StepActorOperation stepActor;
+
+  public Stepping(final StepActorOperation stepActor) {
+    this.stepActor = stepActor;
+  }
+  /**
+   * Return the stepping operation that has been made for a specific source section,
```

Comment:
no, please don't look anymore, now I'm looking at the different scenarios with the breakpoints and I'm changing the implementation. I think the principal problem I had is that I did not consider the stepping for each actor/activity. Thanks for your comments, they have been helpful.